### PR TITLE
Refactored ProtectedMethod pattern

### PR DESCRIPTION
**Pull request by Vitaly-Protasov**
Solved in case of #528

### CODE REVIEW

**Comment by acheshkov** [comment]
@rultor merge

File: aibolit/patterns/protected_method/protected_method.py [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
+from typing import List
+from aibolit.ast_framework.ast_node import ASTNode
 
 
 class ProtectedMethod:
+    '''
+    Once we find a protected method in a class, it's a pattern.
+    '''
+    def _check_protected(self, node: ASTNode) -> bool:
+        if all(type in node.modifiers for type in ['protected']):
```
Comment (aravij) [inline aibolit/patterns/protected_method/protected_method.py:33]: You can simply return result of `all`.

Comment (aravij) [reply aibolit/patterns/protected_method/protected_method.py:33]: It seams like you simply looking if `'protected'` is in modifiers. You may check it directly.

Comment (Vitaly-Protasov) [reply aibolit/patterns/protected_method/protected_method.py:33]: @aravij  What do you mean by checking it directly?

Comment (aravij) [reply aibolit/patterns/protected_method/protected_method.py:33]: something like this
```python
return 'protected' in node.modifiers
```

Comment (Vitaly-Protasov) [reply aibolit/patterns/protected_method/protected_method.py:33]: got it

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Comment by acheshkov** [comment]
@rultor merge

**Pull request merged**